lib/Logger.ts

```diff
@@ -1,241 +1,179 @@
-import {
-	createLogger,
-	transports,
-	transport as Transport,
-	Logger as WinstonLogger
-} from 'winston';
-import 'winston-mongodb';
-import ipc from 'node-ipc';
-import { Ipc } from '../cli/util/enum/Ipc';
-import { Message, Socket } from '../cli/util/interface/Message';
-import { Text, Json } from './format';
-import { host, includes, color, stringify } from './util/Helpers';
-import LoggerConfigOptions from './util/interface/LoggerConfigOptions';
+import { getLogWrapper } from './format';
+import { LoggerMethods } from './util/interface/LoggerMethods';
+import { LogWrapper } from './util/type/LogWrapper';
+import { AbstractLogger } from './util/type/AbstractLogger';
+import { registerEvent } from './util/Events';
+import { LogLevels } from './util/enum/LogLevels';
+import { setupLogServer } from './http/server';
+import { Server } from 'http';
+import { level, output, host } from './util/Helpers';
+import { LevelTags } from './util/enum/LevelTags';
 
 /**
- * OZLogger module class.
+ * Logger module class.
  */
-export class OZLogger {
-	/**
-	 * Stores the Logger object instance.
-	 */
-	private static instance: OZLogger;
-
+export class Logger implements LoggerMethods {
 	/**
 	 * Temporary storage for timers.
 	 */
-	private static timers = new Map<string, number>();
+	private timers = new Map<string, number>();
 
 	/**
-	 * Temporary storage for log tags.
+	 * Temporary storage for timeouts.
 	 */
-	private static tags?: string[];
+	private timeouts = new Map<string, NodeJS.Timeout>();
```

Comment:
This one closely relates to the RPC calls handled by the internal events.

This property is important so that I can keep track if someone called a `/changeLevel` action twice. In  that case, the first request must be canceled (due to duration timeout) in order to keep track of the new log level that was set by the second action call.